elftool/elftool.c

```diff
@@ -210,18 +213,66 @@ static int mfttool_dump(const char *binary)
         return EXIT_FAILURE;
     }
 
-    printf("%s: Manifest contains %d entries.\n", binary, mft->entries);
+    printf("{\n"
+	   "    \"version\": %u,\n"
+	   "    \"devices\": [\n", MFT_VERSION);
+
     for (unsigned i = 0; i != mft->entries; i++) {
-        printf("%s[%u]: Name: '%s', Type: %s\n", binary, i,
-                (mft->e[i].type >= MFT_RESERVED_FIRST) ? "N/A" : mft->e[i].name,
-                mft_type_to_string(mft->e[i].type));
+	if (mft->e[i].type >= MFT_RESERVED_FIRST)
+	    continue;
+
+	printf("        { \"name\": \"%s\", \"type\": \"%s\" }%s\n",
+		mft->e[i].name, mft_type_to_string(mft->e[i].type),
+		(i == (mft->entries -1) ? "" : ","));
     }
 
+    printf("    ]\n"
+	   "}\n");
+
     free(mft);
     close(bin_fd);
```

Comment:
I guess we can `close(bin_fd);` immediately after `elf_load_note()`, then we don't have to keep it around for the rest of the function (and the close on line `211` becomes redundant).